main.js

```diff
@@ -92,24 +92,36 @@ function getMainMapAddrs(json) {
     return addrs;
 }
 
-function drawMainMap(addrList) {
-    let map = L.map('main-map', {
-        center: [44.4781994, -73.2126357],
-        zoom: 20
-    });
+async function drawMainMap(addrList) {
+    let map = L.map('main-map').setView([44.4781994, -73.2126357],15);
 
     L.tileLayer('https://{s}.tile.openstreetmap.fr/osmfr/{z}/{x}/{y}.png', {
         maxZoom: 20,
         attribution: '&copy; Openstreetmap France | &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
     }).addTo(map);
 
-    addrList.forEach((ad) => {
-        const latArr = getCoords(ad.addr);
 
-        console.log(latArr);
-        let mark = L.marker([latArr.lat, latArr.lng]).addTo(map);
-        return mark;
-    })
+    for await (const ad of addrList) {
```

Comment:
here we need to do some funky stuff that I don't think we've gone over in class. If you want to use a loop to make multiple fetches, then you need to do this ```for await...of``` loop, in which you can await another function.